laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -579,278 +426,114 @@ private function insertOption()
-        $attending = substr(
-            $attending_string,
-            strpos($attending_string, "[") + 1,
-            strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-        );
-
-        if (isset($_REQUEST["pref_anest1"])) {
-            // if they chose an anesthesiologist, add their ID to the DB, if not, add NULL
-            if ($_REQUEST["pref_anest1"] != 0) {
-                $pref_anest1 = $_REQUEST["pref_anest1"];
-            }
-        }
-
-        // Update or insert option 1 data
-        if (
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->count() != 0
-        ) {
-            // generate notification and delete data
-            $notify = true;
-            $overwrittenChoices[0] = 1;
-
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->update([
-                    "schedule" => $split[0],
-                    "attending" => $attending,
-                    "milestones" => $_REQUEST["milestones1"],
-                    "objectives" => $_REQUEST["objectives1"],
-                    "anesthesiologist_id" => $pref_anest1,
-                    "isValid" => 1,
-                ]);
-        } else {
-            $overwrittenChoices[0] = 0;
-            // Insert data
-            Option::insert([
-                "date" => $date,
-                "resident" => $resident,
-                "schedule" => $split[0],
-                "attending" => $attending,
-                "option" => $choice,
-                "milestones" => $_REQUEST["milestones1"],
-                "objectives" => $_REQUEST["objectives1"],
-                "anesthesiologist_id" => $pref_anest1,
-                "isValid" => 1,
-            ]);
-        }
-
-        //Update second choice data
-        $choice++;
-        $schedule_data2 = null;
-        $attending = null;
-        // Insert second choice data if it exists
-        if ($split[1] != 0) {
-            // Get schedule data of the 2nd preference
-            $schedule_data2 = ScheduleData::where("id", $split[1])->get();
-            // Get attending id
-            $attending_string = $schedule_data2[0]["lead_surgeon"];
-            $attending = substr(
-                $attending_string,
-                strpos($attending_string, "[") + 1,
-                strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-            );
-        }
-
-        if (isset($_REQUEST["pref_anest2"])) {
-            if ($_REQUEST["pref_anest2"] != 0) {
-                $pref_anest2 = $_REQUEST["pref_anest2"];
-            }
-        }
+        $schedule_data[0] = ScheduleData::where("id", $schedule_data_ids[0])->get();
+        $lead_surgeon_string = $schedule_data[0][0]["lead_surgeon"];
+        $date = $schedule_data[0][0]["date"];
```

Comment:
I believe these all get overwritten within the loop. Do they serve some purpose that I'm missing?

Reply:
You're right, I think I copied and pasted this with the lines above it. Should be out next push!